File: cloud-run-function-kotlin/app/src/main/kotlin/org/phdezann/energy/formater/TimeFormatter.kt
package org.phdezann.energy.formater

import org.phdezann.energy.support.UnitFormatter
import org.phdezann.energy.support.FormatterUtils.toUnits

class TimeFormatter {

    fun format(valueInHours: Double): String {
        return when {
            valueInHours >= 1 -> toUnits(valueInHours, "h")
            valueInHours >= 1.0 / 60 -> toUnits(valueInHours * 60, "m")
            else -> UnitFormatter.format(valueInHours * 3600, false) + "s"
        }
    }

}

File: src/energy-converter/time-formatter.ts
class UnitFormatter {
  static format(value: number, showUnit: boolean): string {
    return showUnit ? `${value.toFixed(2)} units` : value.toFixed(2);
  }
}

class FormatterUtils {
  static toUnits(value: number, unit: string): string {
    return `${value.toFixed(2)} ${unit}`;
  }
}

class TimeFormatter {
  format(valueInHours: number): string {
    if (valueInHours >= 1) {
      return FormatterUtils.toUnits(valueInHours, 'h');
    } else if (valueInHours >= 1.0 / 60) {
      return FormatterUtils.toUnits(valueInHours * 60, 'm');
    } else {
      return UnitFormatter.format(valueInHours * 3600, false) + 's';
    }
  }
}

export default TimeFormatter;
